test/functions/test_retrieval_function.py

```diff
@@ -0,0 +1,103 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+
+import os
+import shutil
+from unittest.mock import MagicMock, Mock, patch
+
+import pytest
+
+from camel.embeddings import OpenAIEmbedding
+from camel.retrievers.retrieval_module import RetrievalModule
+from camel.storages.vectordb_storages import QdrantStorage
+
+
+@pytest.fixture
+def temp_storage_path():
+    # Define the path to the temporary storage
+    path = 'test/functions/tempory_storage'
+
+    # Setup phase: yield the path for use in tests
+    yield path
+
+    # Teardown phase: remove the files created in the temporary storage
+    if os.path.exists(path):
+        shutil.rmtree(path)
+
+
+# Fixtures for common mock objects
+@pytest.fixture
+def mock_openai_embedding():
+    return Mock(spec=OpenAIEmbedding)
+
+
+@pytest.fixture
+def mock_qdrant_storage():
+    return Mock(spec=QdrantStorage)
+
+
+@pytest.fixture
+def retrieval_function_instance():
+    return RetrievalModule()
+
+
+def test_retrieval_function_initialization():
+    retrieval_func = RetrievalModule()
+    assert isinstance(retrieval_func.embedding_model, OpenAIEmbedding)
+
+
+def test__initialize_qdrant_storage_with_valid_path(
```

Comment:
Here it looks like you're using two "_"'s after "test".

Reply:
Hey @FUYICC , thanks for the detailed review!

The reason why I used two "_"'s after "test" is that the object for this test is `_initialize_qdrant_storage`, start with  "_" means this function is for internal use~